97067 disability benefits nb workflow choice page skeleton

NB28VT opened the pull request:
## Are you removing, renaming or moving a folder in this PR?
- [X] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [X] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

Adds a new "Workflow Choice" page to the incoming improved supplemental Form 0781 PTSD workflow for users filling out a Form 526. This PR closes both the [page skeleton](https://github.com/department-of-veterans-affairs/va.gov-team/issues/97067) and [page content/formatting](https://github.com/department-of-veterans-affairs/va.gov-team/issues/97074) tickets.

This page presents a user who has identified PTSD/trauma/SA as a claimed condition(s) with three radio button options:

1. Complete a 0781 form online.
2. Upload an already-completed paper version of Form 0781
3. Opt out of the Form 0781 entirely (some users may want to avoid being re-traumatized by reliving the experience)

A selection is required before the user clicks "continue". When they do, they are redirected to the follow pages from the options above respectively:
1. A page summarizing mental health resources they may want to consult before filling out the form (implemented in https://github.com/department-of-veterans-affairs/vets-website/pull/33811)
2. A Form 0781 PDF upload page (to be implemented in https://github.com/department-of-veterans-affairs/va.gov-team/issues/97066)
3. The next step in the Form 526 flow after the optional Form 0781 flow; in this case it's the supporting evidence upload page at `http://localhost:3001/disability/file-disability-claim-form-21-526ez/supporting-evidence/orientation`

Additionally, adds a placeholder for 2., the yet-to-implement upload page described in https://github.com/department-of-veterans-affairs/va.gov-team/issues/97066

## Related issue(s)

* [Page skeleton](https://github.com/department-of-veterans-affairs/va.gov-team/issues/97067)
* [Page content](https://github.com/department-of-veterans-affairs/va.gov-team/issues/97074)
* [Dynamic New Conditions List and Updated Copy](https://github.com/department-of-veterans-affairs/va.gov-team/issues/101227)

## Testing done

- _Describe what the old behavior was prior to the change_

This page was initially implemented as a placeholder; the entire new Form 0781 flow is under development and hidden behind feature flags.

- _Describe the steps required to verify your changes are working as expected_

1. In the development environment, ensure that the `sync_modern_0781_flow` toggle is enabled
2. Start a new claim
3. Claim at least 1 new condition
4. Select at least 1 of the new conditions on the [0781 screener page](https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=9250-79794)
5. Visit the Form 0781 start page (currently a placeholder) at `http://localhost:3001/disability/file-disability-claim-form-21-526ez/additional-forms` and click "Continue"
6. The next page should be the workflow choice page (`http://localhost:3001/disability/file-disability-claim-form-21-526ez/additional-forms/mental-health-statement`)
7. Confirm the selections route the user to the correct page by making a selection and clicking "Continue"

- _Describe the tests completed and the results_

Confirmed in manual testing:
1. You cannot continue past this page unless you make a selection
2. Making a selection from each of the options and clicking "Continue" routes you to the correct pages as noted above.


Automated testing:
1. The standard page specs have been implemented as was done for the other changes in this flow.
2. A concious decision was made NOT to unit test the new utility function in `src/applications/disability-benefits/all-claims/utils/form0781.js`; while I feel like this conditional routing should be tested, this proved difficult to mock with our testing tools, none of the other functions in that file have automated unit tests, and it was agreed this would be covered when larger integration tests were completed that cover the entire page flow, with associated routing.


## Screenshots

### Choice Page Before:

<img width="482" alt="Screenshot 2025-01-09 at 1 38 16 PM" src="https://github.com/user-attachments/assets/dfb8e0fd-af07-488e-988b-c8b50be7b21e" /> 
 
### Choice Page after (long page, two screenshots)

<img width="464" alt="Screenshot 2025-01-21 at 1 17 47 PM" src="https://github.com/user-attachments/assets/8aaa99a6-3271-4bc5-85af-a84dd3070fe4" />


 <img width="283" alt="Screenshot 2025-01-17 at 11 51 21 AM" src="https://github.com/user-attachments/assets/5197309d-79a6-4897-9594-53184ef388ee" />

### Upload Page Placeholder

<img width="690" alt="Screenshot 2025-01-17 at 11 55 02 AM" src="https://github.com/user-attachments/assets/6619afe1-d703-4fb0-b01a-c096e1f5d033" />

## What areas of the site does it impact?

- Only impacts 526 EZ (within a new "Additional Forms" chapter)

## Acceptance criteria

### Quality Assurance & Testing

- [X] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary) N/A
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed **(linked page does not exist)**

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

## Requested Feedback

Will be noted inline in the PR diff itself

NB28VT committed:
WORKING COMMIT

NB28VT committed:
WORKING COMMIT cleaning up changes

NB28VT commented on src/applications/disability-benefits/all-claims/config/form0781/index.js:
```diff
@@ -13,4 +19,18 @@ export const form0781PagesConfig = {
+    depends: formData => showManualUpload0781Page(formData),
+    schema: manualUploadPage.schema,
+  },
+  mentalHealthSupportInfoPage: {
```
Potentially take out if Scott's merges first (https://github.com/department-of-veterans-affairs/va.gov-team/issues/97078)

NB28VT replied:
Because he will have this page actually implemented

NB28VT replied:
TODO: do unit tests for these: **showManualUpload0781Page(formData),**

NB28VT commented on src/applications/disability-benefits/all-claims/pages/form0781/workflowChoicePage.js:
```diff
@@ -1,9 +1,35 @@
+    type: 'object',
+    required: ['pTSDWorkflowChoice'],
+    properties: {
+      pTSDWorkflowChoice: {
```
TODO: make this a "view" property

NB28VT replied:
Leave at higher level, or consider nesting in 0781, but probably leave at high level as a view property

freeheeling commented on src/applications/disability-benefits/all-claims/config/form0781/index.js:
```diff
@@ -13,4 +19,18 @@ export const form0781PagesConfig = {
     uiSchema: workflowChoicePage.uiSchema,
     schema: workflowChoicePage.schema,
   },
+  manualUploadPage: {
+    path:
+      'disability/file-disability-claim-form-21-526ez/additional-forms/mental-health-statement/upload',
+    uiSchema: manualUploadPage.uiSchema,
+    depends: formData => showManualUpload0781Page(formData),
+    schema: manualUploadPage.schema,
+  },
```
This looks like it's referring to [[3.3], the Upload 0781 page](https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=9442-62961&t=OwUuVSYswnhT7tYQ-4), when I thought the corresponding page for [#97067](https://github.com/department-of-veterans-affairs/va.gov-team/issues/97067) was [[3.2], the 0781 choice page](https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=9442-72041&t=OwUuVSYswnhT7tYQ-4).

freeheeling replied:
Oh, never mind. I realize now that the `workflowChoicePage` is [[3.2], the 0781 choice page](https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=9442-72041&t=OwUuVSYswnhT7tYQ-4).

freeheeling commented on src/applications/disability-benefits/all-claims/pages/form0781/workflowChoicePage.js:
```diff
@@ -1,9 +1,35 @@
+export default {
+  uiSchema: {
+    'ui:title': workflowChoicePageTitle,
+    pTSDWorkflowChoice: radioUI({
```
Perhaps consider a different name as the respective question is related to mental health and not PTSD specifically.

NB28VT replied:
Good suggestion, changed to `mentalHealthWorkflowChoice`

freeheeling commented on src/applications/disability-benefits/all-claims/config/form0781/index.js:
```diff
@@ -1,5 +1,11 @@
+import {
+  showForm0781Pages,
+  showManualUpload0781Page,
+  showmentalHealthSupportInfoPage,
```
Capitalize 'mental' for consistency.

NB28VT replied:
These import names have changed since this comment was originally posted so I think we're good now

NB28VT committed:
WORKING COMMIT Still trying to figure out tests

NB28VT committed:
Get view field working with current web component patterns

freeheeling commented on src/applications/disability-benefits/all-claims/content/form0781.jsx:
```diff
@@ -1 +1,20 @@
+export const workflowChoicePageDescription =
+  'Some stuff about maiking a selection la la la la la la ';
+
+export const form0781WorkflowChoices = {
+  COMPLETE_ONLINE_FORM: 'optForOnlineForm0781',
+  SUBMIT_PAPER_FORM: 'optForPaperForm0781Upload',
+  OPT_OUT_OF_FORM0781: 'optOutOfForm0781',
+};
```
I utilized this in https://github.com/department-of-veterans-affairs/vets-website/pull/33741

NB28VT committed:
Clean up latest iteration of tests to see why not working

NB28VT committed:
Remove conditional routing utility unit test

Unit testing this functionality was proving problematic with Sinon; it was decided to wait to test the conditional routing based on worlflow choice in proper end to end integration tests instead. These will be implemented later in the project

NB28VT committed:
Merge branch 'main' into 97067-disability-benefits-nb-workflow-choice-page-skeleton

# Conflicts:
#	src/applications/disability-benefits/all-claims/config/form0781/index.js
#	src/applications/disability-benefits/all-claims/content/form0781.jsx
#	src/applications/disability-benefits/all-claims/utils/form0781.js

NB28VT committed:
Remove placeholder mental health info page it was replaced with implementation already

NB28VT committed:
Fix view property requirement syntax to get tests passing again

NB28VT committed:
Fix syntax on property key in utility function for workflow choice

NB28VT committed:
Merge branch 'main' into 97067-disability-benefits-nb-workflow-choice-page-skeleton

# Conflicts:
#	src/applications/disability-benefits/all-claims/config/form0781/index.js

NB28VT committed:
Add heading tag with form name

NB28VT committed:
Add page description

NB28VT committed:
Add choices description

NB28VT committed:
Add selection labels

NB28VT committed:
Fix formatting on header

NB28VT committed:
Add collapsable additional info on questions to be asked

NB28VT committed:
Fix broken routing

Checking the form data in the utility functions to see what workflow choice was selected got altered by mistake during the master merge

Additionally, the initial implementation of the routing for the consent page was unaware of the isCompletingForm0781 check (opt in), so it needed to be added for that page

NB28VT committed:
Fix broken test

NB28VT committed:
Merge branch 'main' into 97067-disability-benefits-nb-workflow-choice-page-skeleton

NB28VT commented on src/applications/disability-benefits/all-claims/config/form0781/index.js:
```diff
@@ -51,7 +60,7 @@ export const form0781PagesConfig = {
   additionalInformationPage: {
     path: 'additional-forms/mental-health-statement/additional-information',
-    depends: formData => showForm0781Pages(formData),
+    depends: formData => isCompletingForm0781(formData),
```
I think this page merged before `isCompletingForm0781` was a thing; regardless, this change fixed accidentally jumping to this page from one of the workflow choices, since we should only see this page if the user has opted into the 0781 flow

ajones446 replied:
Yep, I updated that in [PR 33975](https://github.com/department-of-veterans-affairs/vets-website/pull/33975/files). It just got merged into main this morning.

NB28VT commented on src/applications/disability-benefits/all-claims/content/form0781.jsx:
```diff
@@ -4,12 +4,6 @@ import { formTitleTag, formTitle } from '../utils';
 export const form0781HeadingTag = 'VA FORM 21-0781';
 export const additionalFormsTitle = 'Additional Forms';
 
-export const form0781WorkflowChoices = {
```
Created a separate content file for the workflow choice page as we've been doing with the other pages.

NB28VT commented on src/applications/disability-benefits/all-claims/pages/form0781/workflowChoicePage.js:
```diff
@@ -1,9 +1,47 @@
+  uiSchema: {
+    'ui:title': titleWithTag(workflowChoicePageTitle, form0781HeadingTag),
+    'ui:description': workflowChoicePageDescription,
+    'view:mentalHealthWorkflowChoice': radioUI({
```
Note this property is view only, since making this selection is used for page routing, it isn't submitted to the backend

NB28VT commented on src/applications/disability-benefits/all-claims/pages/form0781/workflowChoicePage.js:
```diff
@@ -1,9 +1,47 @@
+      labels: form0781WorkflowChoiceLabels,
+      enableAnalytics: true,
+    }),
+    'view:ptsdQuestionsPreview': {
```
This property is required to use the collapsable content component as displayed in the Figma wireframe:

<img width="400" alt="Screenshot 2025-01-09 at 2 10 02 PM" src="https://github.com/user-attachments/assets/dee10e1e-edad-46bb-8ab1-26825fce6529" />

freeheeling commented on src/applications/disability-benefits/all-claims/pages/form0781/manualUploadPage.js:
You chose not to include a screenshot for this page?

NB28VT replied:
Should we include it? It's just the placeholder text

freeheeling replied:
Just bringing it up, in case it was something you did/didn't intend to include.

NB28VT replied:
I did end up doing this when I posted the revised screenshots

freeheeling commented on src/applications/disability-benefits/all-claims/content/form0781/workflowChoicePage.jsx:
```diff
@@ -0,0 +1,56 @@
+export const workflowChoicePageDescription = (
+  <>
+    <p>
+      You selected these mental health conditions: <b>PTSD, Depression</b>.
```
This should display selections from [2.1] 0781 screener, `formData.mentalHealth.conditions`.

NB28VT replied:
Implemented in https://github.com/department-of-veterans-affairs/vets-website/pull/33655/commits/d68a885d16f38a3446cd4a5d5236936ecd7f76bc and https://github.com/department-of-veterans-affairs/vets-website/pull/33655/commits/6fea11d8dac18237e2e89033841b83d7dc47ef7d per the [latest copy revisions](https://github.com/department-of-veterans-affairs/va.gov-team/issues/101227)

freeheeling commented on src/applications/disability-benefits/all-claims/content/form0781/workflowChoicePage.jsx:
```diff
@@ -0,0 +1,56 @@
+});
+
+export const ptsdQuestionsPreview = (
+  <VaAdditionalInfo trigger="We'll ask you questions about these topics">
```
use [curly apostrophe](https://design.va.gov/content-style-guide/punctuation#apostrophes)

```suggestion
  <VaAdditionalInfo trigger='We’ll ask you questions about these topics'>
```

NB28VT replied:
Fixed in all places, thanks for using the suggested change feature

freeheeling commented on src/applications/disability-benefits/all-claims/tests/pages/form0781/workflowChoicePage.unit.spec.js:
I would also suggest including a test that confirms the correct condition(s) selected on the screener page are rendered in the description.

NB28VT replied:
This should be covered well enough, have a look

freeheeling reviewed (changes requested):
Please review comments.

NB28VT committed:
Merge branch 'main' into 97067-disability-benefits-nb-workflow-choice-page-skeleton

# Conflicts:
#	src/applications/disability-benefits/all-claims/config/form0781/index.js

NB28VT committed:
Implement dynamic mental health condition list

Adds a previously missed requirement to dynamically list the mental health conditions selected on the mental health screener page when making selection on what 0781 flow the user wants to complete

NB28VT committed:
Refactor conditions list for new spec

Updates the conditions list display on the 0781 workflow choice page to list all new disabiites claimed by the user per the latest spec

NB28VT committed:
Add space before conditions list

Fixes the formatting in the conditions list display on the 0781 workflow choice page to include a space befor ethe list

NB28VT committed:
Use curly format apostrophe

NB28VT committed:
Update existing workflow choice copy for new revisions

NB28VT committed:
Replace existing additional info dropdown with mental health condition accordion card

NB28VT committed:
Merge branch 'main' into 97067-disability-benefits-nb-workflow-choice-page-skeleton

NB28VT committed:
Fix outdated content import

NB28VT commented:
TODO: tests should take care of this, but as pointed out in https://github.com/department-of-veterans-affairs/va.gov-team/issues/101227 we should make sure if the user goes back and changes their selection of new disabilities it is presented on the new page (and the page is not cached, the forms library should ideally take care of this)

NB28VT commented:
> TODO: tests should take care of this, but as pointed out in [department-of-veterans-affairs/va.gov-team#101227](https://github.com/department-of-veterans-affairs/va.gov-team/issues/101227) we should make sure if the user goes back and changes their selection of new disabilities it is presented on the new page (and the page is not cached, the forms library should ideally take care of this)

Confirmed this locally

freeheeling commented on src/applications/disability-benefits/all-claims/content/form0781/workflowChoicePage.jsx:
```diff
@@ -0,0 +1,156 @@
+      <p>
+        {/* Leave space ahead of conditions list */}
+        You selected these new conditions for your disability claim:{' '}
+        <strong>{conditions.join(', ')}</strong>.
```
I know that design has comma separates conditions but, after viewing the screenshot provided in the PR description, I wonder if that choice should be reconsidered as there are conditions that also include commas and, when joined with other conditions, the differentiation may be confusing. Just mentioning for consideration.

NB28VT replied:
Yeah I noticed that as well meant to mention that in the PR, it's less confusing because there the conditions are capitalized but it's still confusing. I'm not how we get around that, unless we revise this and list the conditions in bullet points, I can ask Ruben about that

NB28VT replied:
Yes we will do bulleted list here

NB28VT replied:
Made the change for this [here](https://github.com/department-of-veterans-affairs/vets-website/pull/33655/commits/320f881724502c90482556dc04628e209e440c6e) and updated the screenshot in the PR description

NB28VT committed:
Change dynamic condition list to ul element

Veteran-selected conditions that are listed on the Form 0781 workflow choice page can sometimes including commas, this can be confusing in a paragraph element when there are multiple conditions, so we are switching to a bulleted list

freeheeling commented on src/applications/disability-benefits/all-claims/content/form0781/workflowChoicePage.jsx:
```diff
@@ -0,0 +1,159 @@
+    'No, I don’t want to complete VA Form 21-0781 (opt out)',
+});
+
+export const traumaticEventsExamples = (
```
This content is also in src/applications/disability-benefits/all-claims/content/form0781.jsx.

NB28VT replied:
Not quite, this one differs slightly and reflects the latest content spec from Ruben for this specific page. Also I think we're removing the other cases in a separate ticket?

Regardless, I'd say this is fine here because it's specific to this page and imported as such

freeheeling replied:
I stand corrected. Didn't look closely enough, obviously.

freeheeling commented on src/applications/disability-benefits/all-claims/content/form0781/workflowChoicePage.jsx:
```diff
@@ -0,0 +1,159 @@
+      disability.condition.slice(1),
+  );
+
+  if (conditions?.length) {
+    return (
+      <div>
+        <p>You selected these new conditions for your disability claim:</p>
+        <ul>
+          {conditions.map((condition, index) => (
+            <li key={index}>{condition}</li>
+          ))}
+        </ul>
+      </div>
+    );
+  }
+
+  return <></>;
```
Shouldn't there always be at least 1 condition for this page to render?

NB28VT replied:
Yes, this is pretty defensive, noted this [here](https://github.com/department-of-veterans-affairs/vets-website/pull/33655/files/320f881724502c90482556dc04628e209e440c6e#diff-5485bdc8546bee34e56df206822fac64a4da9aff823504a10efc54a53f6c0323R97) in the test.

I would assume we would want this page to render with the other helpful information instead of blowing up if there is a bug that causes the conditions to be missing and still allow getting to this page. What do you think? Maybe we can ask Ruben and Lisa about this

freeheeling replied:
I did notice your inclusion of that test, but was also wondering if the page shouldn't be rendered at all. Not sure what protocol is in this situation.

NB28VT replied:
Yeah revisiting this and the [content that follows it ](https://github.com/department-of-veterans-affairs/vets-website/pull/33655/files#diff-069586937c2b741226c820e0f34bc68962a1ed01a5194b71bfde4d334bdfa2ccR39) I think the page copy would never make a whole lot of sense without the conditions and we would want it to blow up. 

So I think I am going to remove the guard and the associated test here

NB28VT replied:
Did this in https://github.com/department-of-veterans-affairs/vets-website/pull/33655/commits/9495de49b142380baff48cbff69e3e22e724e0e8

freeheeling commented on src/applications/disability-benefits/all-claims/tests/pages/form0781/workflowChoicePage.unit.spec.js:
```diff
@@ -1,12 +1,212 @@
+    ).not.to.exist;
+  });
+
+  it('Displays a selection of choices on filling out 0781 and incplues examples', () => {
```
> and incplues examples

typo? "incplues" ➡ "includes"

freeheeling commented on src/applications/disability-benefits/all-claims/tests/pages/form0781/workflowChoicePage.unit.spec.js:
```diff
@@ -1,12 +1,212 @@
+    );
+  });
+
+  it('should prevent continuing if a selection is not made', () => {
```
Is an error message displayed if attempting to continue from this page without making a selection? Should the content of that error message be tested?

NB28VT replied:
Yeah this is a good call, figured this out in https://github.com/department-of-veterans-affairs/vets-website/pull/33655/commits/5475dff3a6c59f00f7e8606a7ecef76ee44439a9

freeheeling commented on src/applications/disability-benefits/all-claims/utils/form0781.js:
```diff
@@ -40,6 +40,14 @@ export function showForm0781Pages(formData) {
   );
 }
 
+export function showManualUpload0781Page(formData) {
```
I didn't see a test for this function.

NB28VT replied:
Yeah remember we discussed these utility functions and agreed anything relating to the page routing was best saved for e2e tests? This one ended up being a [real rabbit hole](https://dsva.slack.com/archives/C053U7BUT27/p1734729884214569) because of the need to do some stubbing our test library doesn't easily support. I spent a few days trying to get that to work and stopped because we discussed it and decided we could cover it later with e2e tests.

NB28VT committed:
PR feedback: fix docstring typo

NB28VT committed:
Remove defensive guard clause on conditions list

I had initially implemented this page in a way where some of the content would still render if for some reason the list of newly claimed conditions was missing. The user should not be able to get to this page if that has happened; therfore it is preferable for this page to blow up instead of rendering incomplete content

NB28VT committed:
Add more robust test for error message

Add test for error message content

NB28VT committed:
Merge branch 'main' into 97067-disability-benefits-nb-workflow-choice-page-skeleton

freeheeling commented on src/applications/disability-benefits/all-claims/config/form0781/index.js:
```diff
@@ -34,6 +36,13 @@ export const form0781PagesConfig = {
     schema: workflowChoicePage.schema,
   },
+  manualUploadPage: {
+    path:
```
Probably only need the subpath, `additional-forms/mental-health-statement/upload`, and not the full path here, when this page is fully implemented.
